src/charm.py

```diff
@@ -203,10 +203,13 @@ def _create_env(self):
         Returns:
             env: dictionary of environment variables
         """
-        if "superset-secret-key" in self.config:
+        if self.config["superset-secret-key"]:
             superset_secret = self.config["superset-secret-key"]
         else:
-            superset_secret = generate_random_string(32)
+            superset_secret = self._state.secret_key or generate_random_string(
+                32
+            )
+        self._state.superset_key = superset_secret
```

Comment:
Note for the future: This should check for unit leadership before setting to state.

Reply:
Noted, I think I'll address this in full when I work on the scaling story.